pandas/core/config_init.py

```diff
@@ -500,7 +500,7 @@ def use_inf_as_na_cb(key) -> None:
         "string_storage",
         "python",
         string_storage_doc,
-        validator=is_one_of_factory(["python", "pyarrow"]),
+        validator=is_one_of_factory(["python", "pyarrow", "pyarrow_numpy"]),
```

Comment:
Do we want the user to allow setting this for the new option as well? Because we also have the new `pd.options.future.infer_strings` option to enable the future string dtype?

Reply:
I'd say yes, you might want to astype columns after operations for example

Reply:
Probably something to further discuss in a follow up issue, but I would expect that if you opt-in for the future string dtype with `pd.options.future.infer_strings`, that this would also automatically set the "pyarrow_numpy" string storage as default for operations that depend on that (like doing `astype("string")`)

Reply:
Opened https://github.com/pandas-dev/pandas/issues/54793 for this interaction between `pd.options.future.infer_strings` and the default `string_storage`